File: solutions/2938. Separate Black and White Balls/2938.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def minimumSteps(selff, s: str) -> int:
    ans = 0
    ones = 0

    for c in s:
      if c == '1':
        ones += 1
      else:
        ans += ones

    return ans

File: solutions/2938. Separate Black and White Balls/2938.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  long long minimumSteps(string_view s) {
    long ans = 0;
    int ones = 0;

    for (const char c : s)
      if (c == '1')
        ++ones;
      else
        ans += ones;

    return ans;
  }
};